htcrystalball/collect.py

```diff
@@ -1,106 +1,48 @@
-    except ValueError:
-        return None
-
-
-def format_slots(slots: list) -> dict:
-    """
-    Reassign slot values to new keys for compatibility.
-
-    The dictionary of slots is formatted into a new dictionary similar to the
-    already used key:value pairs in HTCrystalball.
-    """
-    formatted = {"slots": []}
-
-    for slot in slots:
-        # TODO: Check if float conversion is necessary
-        name = slot['UtsnameNodename']
-        n_cpus = int(slot.get('TotalSlotCpus', 0))
-        n_gpus = int(slot.get('TotalSlotGPUs', 0))
-        n_slots = int(slot.get('TotalSlots', 0))
-        disk = float(slot.get('TotalSlotDisk', 0.0))
-        ram = float(slot.get('TotalSlotMemory', 0.0))
-        slot_type = slot['SlotType']
-
-        slot_size = {
-            "TotalSlotCpus": n_cpus,
-            "TotalSlotDisk": kib_to_gib(disk),
-            "TotalSlotMemory": mib_to_gib(ram)
-        }
-
-        if slot_type == "Partitionable":
-            if "gpu" in name and n_gpus != 0:
-                slot_size["SlotType"] = "gpu"
-                slot_size["TotalSlotGPUs"] = n_gpus
-            else:
-                slot_size["SlotType"] = "partitionable"
-                slot_size["TotalSlotGPUs"] = 0
-
-        elif slot_type == "Static":
-            slot_size["SlotType"] = "static"
-        else:
-            continue
-
-        slot_size["TotalSlots"] = n_slots
-        node_in_list = node_name_in_list(
-            slot["UtsnameNodename"],
-            formatted["slots"]
-        )
-
-        if node_in_list is not None:
-            if slot_size not in formatted["slots"][node_in_list]["slot_size"]:
-                formatted["slots"][node_in_list]["slot_size"].append(slot_size)
-        else:
-            formatted_slot = {
-                'UtsnameNodename': name,
-                'slot_size': [slot_size]
-            }
-            formatted["slots"].append(formatted_slot)
-
-    return formatted
-
-
 def collect_slots(filename: Union[str, None] = None) -> dict:
     """Get the condor config and creates a dict."""
-    status = {"slots": []}
-    projection = [
-        "SlotType", "UtsnameNodename", "Name", "TotalSlotCpus",
-        "TotalSlotDisk", "TotalSlotMemory", "TotalSlots", "TotalSlotGPUs"
-    ]
+    projection = ["SlotType", "UtsnameNodename", "TotalSlotCpus", "TotalSlotDisk", "TotalSlotMemory", "TotalSlots",
+                  "TotalSlotGPUs"]
+    unique_slots = {}
 
     if filename is None:
         coll = htcondor.Collector()
-        content = coll.query(
-            htcondor.AdTypes.Startd,
-            projection=projection
-        )
+        content = coll.query(htcondor.AdTypes.Startd, constraint='SlotType != "Dynamic"', projection=projection)
         for slot in content:
-            if "Name" in slot:
-                value = slot["Name"].split('@')[0]
-                slot["Name"] = value.replace("\"", "")
+            nodename = slot['UtsnameNodename']
+            slot_as_dict = {
```

Comment:
There seems to be two instances of `slot_as_dict`. Is that intentional?

Reply:
As I apply mocking, one of them will be removed. In theory now there is only one instance used. One in an active condor pool, the other in tests where there is currently no pool available.